xskillscore/core/contingency.py

```diff
@@ -796,3 +796,223 @@ def _gerrity_s(table):
+            if str(o_check.dtype) != "bool":
+                if not ((o_check == 0) | (o_check == 1)).all():
+                    raise ValueError(
+                        'Input "observations" must represent logical (True/False) outcomes',
+                        o_check,
+                    )
+
+            # works only for 1var
+            if isinstance(forecasts, xr.Dataset):
+                v = list(forecasts.data_vars)[0]
+                f_bin = forecasts[v]
+            else:
+                f_bin = forecasts
+            f_bin = f_bin.stack(ndim=forecasts.dims)
+            f_bin = f_bin.sortby(-f_bin)
+            bin_edges = np.append(f_bin[0] + 1, f_bin)
+            bin_edges = np.unique(bin_edges)[::-1]
+
+    # loop over each bin_edge and get hit rate and false alarm rate from contingency
+    hr, far = [], []
+    for i in bin_edges:
+        dichotomous_category_edges = np.array(
+            [-np.inf, i, np.inf]
+        )  # "dichotomous" means two-category
+        dichotomous_contingency = Contingency(
+            observations,
+            forecasts,
+            dichotomous_category_edges,
+            dichotomous_category_edges,
+            dim=dim,
+        )
+        far.append(dichotomous_contingency.false_alarm_rate())
+        hr.append(dichotomous_contingency.hit_rate())
+    hr = xr.concat(hr, "probability_bin")
+    far = xr.concat(far, "probability_bin")
+
+    easyfillna = True
+    if easyfillna:
+        far = far.fillna(1.0)
+        hr = hr.fillna(0.0)
+
+    # pad (0,0) and (1,1)
+    far_pad = xr.concat(
+        [
+            xr.ones_like(far.isel(probability_bin=0, drop=True)),
+            far,
+            xr.zeros_like(far.isel(probability_bin=-1, drop=True)),
+        ],
+        "probability_bin",
+    )  # .sortby(far)
+    hr_pad = xr.concat(
+        [
+            xr.ones_like(hr.isel(probability_bin=0, drop=True)),
+            hr,
+            xr.zeros_like(hr.isel(probability_bin=-1, drop=True)),
+        ],
+        "probability_bin",
+    )  # .sortby(hr)
+    # far=far.bfill('probability_bin').ffill('probability_bin')
+    # hr=hr.ffill('probability_bin').bfill('probability_bin')
+
+    # https://github.com/scikit-learn/scikit-learn/blob/42aff4e2edd8e8887478f6ff1628f27de97be6a3/sklearn/metrics/_ranking.py#L916
+    # Attempt to drop thresholds corresponding to points in between and
+    # collinear with other points. These are always suboptimal and do not
+    # appear on a plotted ROC curve (and thus do not affect the AUC).
+    # Here np.diff(_, 2) is used as a "second derivative" to tell if there
+    # is a corner at the point. Both fps and tps must be tested to handle
+    # thresholds with multiple data points (which are combined in
+    # _binary_clf_curve). This keeps all cases where the point should be kept,
+    # but does not drop more complicated cases like fps = [1, 3, 7],
+    # tps = [1, 2, 4]; there is no harm in keeping too many thresholds.
```

Comment:
you able to summarize here? i.e. `_binary_clf_curve` doesn't exist in this package.